From 550115412955f6757bb11f05031cb13b636ac545 Mon Sep 17 00:00:00 2001
From: Sam Nazarko <email@samnazarko.co.uk>
Date: Mon, 21 Dec 2015 04:32:14 +0000
Subject: [PATCH] Add sigterm handler. Send a Quit signal if we're not shutting
 down. If we are shutting down, then systemd may trigger our sigterm handler.
 If this happens, we don't want to be running teardown code twice, so let's
 check if we're already shutting down.

Signed-off-by: Sam Nazarko <email@samnazarko.co.uk>
---
 xbmc/platform/posix/main.cpp | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/xbmc/platform/posix/main.cpp b/xbmc/platform/posix/main.cpp
index b6b7035..ce2a211 100644
--- a/xbmc/platform/posix/main.cpp
+++ b/xbmc/platform/posix/main.cpp
@@ -42,6 +42,18 @@
 #endif
 #include "platform/XbmcContext.h"
 
+#include "Application.h"
+#include "messaging/ApplicationMessenger.h"
+
+using namespace KODI::MESSAGING;
+
+void sigterm_handler(int signum)
+{
+       CLog::Log(LOGINFO, "OSMC: received signal to tear down Kodi");
+       if (! g_application.isShuttingDown())
+          CApplicationMessenger::GetInstance().PostMsg(TMSG_QUIT);
+}
+
 #ifdef __cplusplus
 extern "C"
 #endif
-- 
2.7.4

